runtime/tests/checker/typeargument_test.go

```diff
@@ -178,6 +155,79 @@ func TestCheckTypeArguments(t *testing.T) {
 	})
 }
 
+type checkParameterizedTypeIsInstantiatedTestCase struct {
+	name string
+	code string
+}
+
+func TestCheckParameterizedTypeIsInstantiated(t *testing.T) {
+
+	t.Parallel()
+
+	baseValueActivation := sema.NewVariableActivation(sema.BaseValueActivation)
+	baseValueActivation.DeclareValue(stdlib.InclusiveRangeConstructorFunction)
+	options := ParseAndCheckOptions{
+		Config: &sema.Config{
+			BaseValueActivation: baseValueActivation,
+		},
+	}
+
+	runTestCase := func(t *testing.T, testCase checkParameterizedTypeIsInstantiatedTestCase) {
+		t.Run(testCase.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			_, err := ParseAndCheckWithOptions(t,
+				testCase.code,
+				options,
+			)
+
+			errs := RequireCheckerErrors(t, err, 1)
+
+			assert.IsType(t, &sema.MissingTypeArgumentError{}, errs[0])
+		})
+	}
+
+	testCases := []checkParameterizedTypeIsInstantiatedTestCase{
+		{
+			name: "InclusiveRange",
+			code: "let inclusiveRange: InclusiveRange = InclusiveRange(1, 10)",
+		},
+		{
+			name: "VariableSizedArray with InclusiveRange",
+			code: "let r: [InclusiveRange] = []",
+		},
+		{
+			name: "ConstantSizedType with InclusiveRange",
+			code: "let r: [InclusiveRange; 2] = [InclusiveRange(1, 2), InclusiveRange(3, 4)]",
+		},
+		{
+			name: "OptionalType with InclusiveRange",
+			code: "let r: InclusiveRange? = nil",
+		},
+		{
+			name: "DictionaryType with InclusiveRange",
+			code: "let r: {Int: InclusiveRange} = {}",
+		},
+		{
+			name: "Struct with InclusiveRange",
+			code: `
+				pub struct Foo {
+					pub let a: InclusiveRange
```

Comment:
No need for access modifiers in checker tests by default. Also, this makes the code ready for the Cadence 1.0 branch.

```suggestion
				struct Foo {
					let a: InclusiveRange
```